src/components/Quote.js

```diff
@@ -0,0 +1,48 @@
+import React, { useState, useEffect } from 'react';
+
+const Quote = () => {
+  const api = 'https://api.api-ninjas.com/v1/quotes?category=design';
+  const key = 'GIDZtRlO9nOuhs77+bdOLg==6DdOB8VgjtrE29Vt';
+  const [data, setData] = useState({});
+  const [isLoading, setIsLoading] = useState(true);
+  const [hasError, setHasError] = useState(false);
+
+  useEffect(() => {
+    const fetchData = async () => {
+      setIsLoading(true);
+      try {
+        const response = await fetch(api, {
+          headers: {
+            'X-Api-Key': key,
+          },
+        });
+        const data = await response.json();
+        setData(data[0]);
+        setIsLoading(false);
+      } catch (error) {
+        console.error(error);
+        setIsLoading(false);
+        setHasError(true);
+      }
+    };
+    fetchData();
+  }, []);
+
+  if (isLoading) {
+    return <div>Loading...</div>;
+  }
+
+  if (hasError) {
+    return <div>Error while fetching data</div>;
+  }
+
+  return (
+    <div className="quote">
+      <h1>{data.category}</h1>
+      <p>{data.quote}</p>
+      <small>{data.author}</small>
+    </div>
+  );
+};
```

Comment:
[Optional 😣 ]

The code above is working so perfectly well and looks pretty good nevertheless you can take a look at this piece of code also to avoid writing long code and make your code really short dry and meaningful...

useEffect(() => {
    fetch('https://api.api-ninjas.com/v1/quotes?category=', {
      headers: {
        'X-Api-Key': 'your unique Id',
      },
    }).then((response) => response.json()).then((data) => {
      const body = `${data[0].quote}`;
      setQuote(body);
      setLoading(false);
      setError(null);
    }).catch((e) => {
      setError(e.message);
      setLoading(false);
    });
  }, []);

  return (
    <div className="Quote">
      {error && <p>{error}</p>}
      {loading && <p>Loading...</p>}
      {quote && <p>{quote}</p>}
    </div>
  );
}; 

I hope you find this interesting.

Reply:
Hi @TaiwoEnoch,
Thanks 🙏  for reviewing my project and giving nice✨ and constructive feedback 👍, I will consider your suggestion ♻️